src/app/dataset-view/additional-components/dataset-settings-component/tabs/access/dataset-settings-access-tab/add-people-modal/add-people-modal.component.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright Kamu Data, Inc. and contributors. All rights reserved.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the LICENSE file.
+ */
+
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, inject, Input, OnInit } from "@angular/core";
+import { NgbActiveModal, NgbTypeaheadSelectItemEvent } from "@ng-bootstrap/ng-bootstrap";
+import { OperatorFunction, Observable, debounceTime, distinctUntilChanged, tap, finalize, switchMap } from "rxjs";
+import {
+    AccountWithRole,
+    AccountWithRoleConnection,
+    DatasetAccessRole,
+    DatasetBasicsFragment,
+    NameLookupResult,
+} from "src/app/api/kamu.graphql.interface";
+import { BaseComponent } from "src/app/common/components/base.component";
+import AppValues from "src/app/common/values/app.values";
+import { MaybeNull } from "src/app/interface/app.types";
+import { DatasetCollaborationsService } from "../dataset-collaborations.service";
+import { takeUntilDestroyed } from "@angular/core/rxjs-interop";
+import { LoggedUserService } from "src/app/auth/logged-user.service";
+import { ROLE_OPTIONS } from "./add-people-modal.model";
+
+@Component({
+    selector: "app-add-people-modal",
+    templateUrl: "./add-people-modal.component.html",
+    styleUrls: ["./add-people-modal.component.scss"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AddPeopleModalComponent extends BaseComponent implements OnInit {
+    @Input({ required: true }) public datasetBasics: DatasetBasicsFragment;
+    @Input({ required: true }) public collaborator: MaybeNull<AccountWithRole>;
+    public activeCollaboratorsIds: string[] = [];
+    public role: MaybeNull<DatasetAccessRole> = null;
+    public searchPerson: string = "";
+    public selectedCollaborator: MaybeNull<NameLookupResult>;
+    public searching: boolean = false;
+
+    public readonly DatasetAccessRole: typeof DatasetAccessRole = DatasetAccessRole;
+    public readonly SELECT_ROLE_OPTIONS = ROLE_OPTIONS;
+    private readonly DEFAULT_PER_PAGE = 10;
+
+    public activeModal = inject(NgbActiveModal);
+    private cdr = inject(ChangeDetectorRef);
+    private datasetCollaborationsService = inject(DatasetCollaborationsService);
+    private loggedUserService = inject(LoggedUserService);
+
+    public ngOnInit(): void {
+        this.initState();
+    }
+    private initState(): void {
+        if (this.collaborator) {
+            this.role = this.collaborator.role;
+        }
+        this.datasetCollaborationsService
+            .listCollaborators({ datasetId: this.datasetBasics.id })
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((result: AccountWithRoleConnection) => {
+                this.activeCollaboratorsIds = result.nodes.map((node) => node.account.id);
+            });
+    }
```

Comment:
Modal should be getting this data from upper level. Your modal is trying to be too smart.

Reply:
I disagree here. It is better to make this request in a modal. We can not open the modal window and then an extra request will be executed in the top-level component.